backend/lib/services/infrastructureSecrets.js

```diff
@@ -216,28 +216,29 @@ exports.create = async function ({ user, namespace, body }) {
   })
 }
 
-function checkIfOwnSecret (bodySecretBinding) {
-  const secretNamespace = _.get(bodySecretBinding, 'secretRef.namespace')
-  const secretBindingNamespace = _.get(bodySecretBinding, 'metadata.namespace')
+function isOwnSecret (secretBinding) {
+  const secretNamespace = _.get(secretBinding, 'secretRef.namespace')
+  const namespace = _.get(secretBinding, 'metadata.namespace')
 
-  if (secretNamespace !== secretBindingNamespace) {
-    throw new MethodNotAllowed('Patch allowed only for secrets in own namespace')
-  }
+  return secretNamespace === namespace
 }
 
-exports.patch = async function ({ user, namespace, bindingName, body }) {
+exports.patch = async function ({ user, namespace, name, body }) {
   const client = user.client
 
-  const secretBinding = await client['core.gardener.cloud'].secretbindings.get(namespace, bindingName)
+  const secretBinding = await client['core.gardener.cloud'].secretbindings.get(namespace, name)
   const secretName = _.get(secretBinding, 'secretRef.name')
+  const secretNamespace = _.get(secretBinding, 'secretRef.namespace')
```

Comment:
We could also here use `secretRef.namespace` instead of `secretNamespace`

Reply:
✅